source/infraspective/specs/public/AuditPolicy.ps1

```diff
@@ -1,121 +1,100 @@
+                "Network Policy Server", "User / Device Claims", "Group Membership", "File System",
+                "Registry", "Kernel Object", "SAM", "Certification Services", "Application Generated",
+                "Handle Manipulation", "File Share", "Filtering Platform Packet Drop",
+                "Filtering Platform Connection", "Other Object Access Events",
+                "Detailed File Share", "Removable Storage", "Central Policy Staging", "Sensitive Privilege Use",
+                "Non Sensitive Privilege Use", "Other Privilege Use Events", "Process Termination", "DPAPI Activity",
+                "RPC Events", "Plug and Play Events", "Token Right Adjusted Events", "Process Creation",
+                "Audit Policy Change", "Authentication Policy Change", "Authorization Policy Change",
+                "MPSSVC Rule-Level Policy Change", "Filtering Platform Policy Change",
+                "Other Policy Change Events", "User Account Management", "Computer Account Management",
+                "Security Group Management", "Distribution Group Management", "Application Group Management",
+            "Other Account Management Events", "Directory Service Changes", "Directory Service Replication",
+            "Detailed Directory Service Replication", "Directory Service Access",
+            "Kerberos Service Ticket Operations", "Other Account Logon Events",
+            "Kerberos Authentication Service", "Credential Validation"
+            )]
+            [string]$Target,
+
+            # A Script Block defining a Pester Assertion.
+            [Parameter(Mandatory, Position=3)]
         [scriptblock]$Should
     )
-    Function GetAuditPolicy([string]$Category,[string]$Subcategory) {
-        If (Test-RunAsAdmin){
-            auditpol /get /category:$Category |
-                Where-Object -FilterScript {$_ -match "^\s+$Subcategory"} | 
-                    ForEach-Object -Process {($_.trim() -split "\s{2,}")[1]}
-        } Else {
-            Throw "You must run as Administrator to test AuditPolicy"
+    begin {
+        function GetAuditPolicy {
+            <#
+            .SYNOPSIS
+                Retrieve the given audit policy
+            .DESCRIPTION
+                A wrapper around auditpol
+            #>
+            [CmdletBinding()]
+            param(
+                # The category to get
+                [Parameter( Mandatory = $true
+                )]
+                [string]$Category,
+
+                # The subcategory to get
+                [Parameter( Mandatory = $true
+                )]
+                [string]$Subcategory
+                )
+                begin {
+                }
+                process {
+                    if (Test-RunAsAdmin){
+                        $policy = auditpol /get /category:$Category |
+                            Where-Object -FilterScript {$_ -match "^\s+$Subcategory"} |

```

Comment:
Subcategory is clearly used right here.  I'll have to investigate why PSSA isn't seeing it

Reply:
PSSA is not able to determine that `$Subcategory` is the same one as the parameter, because it is inside a scriptblock scope. There are several discussions about this but one reasonable explanation is [here](https://github.com/PowerShell/PSScriptAnalyzer/issues/827).